frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/model/SummarizedFrameworkInfo.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved. 
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING 
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND 
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. 
+
+package com.microsoft.frameworklauncher.common.model;
+
+import java.io.Serializable;
+
+public class SummarizedFrameworkInfo implements Serializable {
```

Comment:
>SummarizedFrameworkInfo [](start = 13, length = 23)

what is the difference between this and FrameworkInfo? This one doesn't have "getAggregatedTaskRoleStatuses"? Can we reuse summarizedframeworkinfo to derive FrameWorkInfo (only add aggregated task role)

Reply:
I will remove SummarizedFrameworkInfo to speed up list.

---
In reply to: [166514831](https://github.com/Microsoft/pai/pull/309#discussion_r166514831) [](ancestors = 166514831)

Reply:
FrameworkInfo contains all info about the Framework.
FrameworkInfo = AggregatedFrameworkRequest + AggregatedFrameworkStatus.

A Framework's SummarizedFrameworkInfo consists selected fields from its Status and Request.
So, it should be FrameworkInfo derives SummarizedFrameworkInfo.
And to avoid deep copy, we derives SummarizedFrameworkInfo directly from FrameworkRequest and FrameworkStatus.

---
In reply to: [166514831](https://github.com/Microsoft/pai/pull/309#discussion_r166514831) [](ancestors = 166514831)